src/UnderlineNav2/UnderlineNavItem.tsx

```diff
@@ -62,13 +62,29 @@ export const UnderlineNavItem = forwardRef(
   ) => {
     const backupRef = useRef<HTMLElement>(null)
     const ref = forwardedRef ?? backupRef
-    const {setChildrenWidth, selectedLink, setSelectedLink, afterSelect, variant} = useContext(UnderlineNavContext)
+    const {
+      setChildrenWidth,
+      setNoIconChildrenWidth,
+      selectedLink,
+      setSelectedLink,
+      afterSelect,
+      variant,
+      iconsVisible
+    } = useContext(UnderlineNavContext)
     const {theme} = useTheme()
     useLayoutEffect(() => {
       const domRect = (ref as MutableRefObject<HTMLElement>).current.getBoundingClientRect()
+      // might want to select this better
+      const icon = (ref as MutableRefObject<HTMLElement>).current.children[0].children[0]
```

Comment:
maybe you can add `data-component`?

Reply:
I tried selecting the icons using data-component  `item.getAttribute('data-component') === 'leadingIcon'` but  icon gets undefined if any item goes into more menu and comes back 🤔 Weird, I'll look into it again but keeping as is for now.